compiler/safetylib/safetyInterpreter.ml

```diff
@@ -1352,6 +1352,19 @@ end = struct
       let e = Papp1 (E.Olnot ws, e1) in
       [Some e]
 
+    | Sopn.Oslh op ->
+       begin match op with
+       | SLHinit -> [ Some (pcast U64 (Pconst (Z.of_int 0))) ]
+       | SLHupdate ->
+          let b, msf = as_seq2 es in
+          let msf = Pif (Bty (U U64), b, msf, pcast U64 (Pconst (Z.of_int (-1)))) in
+          [ Some msf ]
+       | SLHmove -> let msf = as_seq1 es in [ Some msf ]
+       | SLHprotect _ | SLHprotect_ptr _ ->
```

Comment:
This is suspicious, what about protect(x, -1)? I think you assume that the program is well typed for SCT.

Reply:
What’s the semantics of `#protect`?

Reply:
You are right, I forgot that the semantic of protect is the identity at source level.
Sorry for that.

Reply:
This program is well-typed and has a well-defined semantics (the identity). However, it cannot be compiled.

~~~
export
fn p(reg u64 x) -> reg u64 {
reg u64 y msf;
msf = 42;
y = #protect(x, msf);
return y;
}
~~~